Intersect.Editor/Forms/DockingElements/frmMapLayers.cs

```diff
@@ -872,23 +881,44 @@ private void btnRemoveMapNpc_Click(object sender, EventArgs e)
 
                 Core.Graphics.TilePreviewUpdated = true;
             }
+
+            UpdateNpcCountLabel();
+        }
+
+        private void UpdateNpcCountLabel()
+        {
+            lblNpcCount.Text = Strings.NpcSpawns.SpawnCount.ToString(lstMapNpcs.Items.Count);
         }
 
         private void lstMapNpcs_Click(object sender, EventArgs e)
         {
-            if (lstMapNpcs.Items.Count > 0 && lstMapNpcs.SelectedIndex > -1)
+            lstMapNpcs_Update();
+        }
+        
+        private void lstMapNpcs_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            lstMapNpcs_Update();
+        }
+
+        private void lstMapNpcs_Update()
+        {
+            if (lstMapNpcs.Items.Count <= 0 || lstMapNpcs.SelectedIndex <= -1)
             {
-                cmbNpc.SelectedIndex = NpcBase.ListIndex(Globals.CurrentMap.Spawns[lstMapNpcs.SelectedIndex].NpcId);
-                cmbDir.SelectedIndex = (int) Globals.CurrentMap.Spawns[lstMapNpcs.SelectedIndex].Direction;
-                if (Globals.CurrentMap.Spawns[lstMapNpcs.SelectedIndex].X >= 0)
-                {
-                    rbDeclared.Checked = true;
-                }
-                else
-                {
-                    rbRandom.Checked = true;
-                }
+                return;
+            }
+
+            cmbNpc.SelectedIndex = NpcBase.ListIndex(Globals.CurrentMap.Spawns[lstMapNpcs.SelectedIndex].NpcId);
+            cmbDir.SelectedIndex = (int)Globals.CurrentMap.Spawns[lstMapNpcs.SelectedIndex].Direction;
+            if (Globals.CurrentMap.Spawns[lstMapNpcs.SelectedIndex].X >= 0)
```

Comment:
Store these big, repeated calls to 'Globals' and such to vars instead of repeating them to reduce visual noise

Reply:
we had tmpMap stored at the very beggining 🤣 
![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/17498701/30c576ba-b14a-4606-9438-4ffe94346aff)

Reply:
Done !